### ISSUE TITLE
Increase the cases detected by dangerous-default-value (W0102) ?

### ISSUE THREAD

**Issue by JonJagger** [pylint-dev/pylint#9435]
### Current problem

[W0102](https://pylint.pycqa.org/en/latest/user_guide/messages/warning/dangerous-default-value.html) detects a default value such as `[ ]`
For example, from the docs:
```
def whats_on_the_telly(penguin=[]):
    penguin.append("property of the zoo")
    return penguin
```
    
Consider the following example:
```
import time
def whats_on_the_telly(timestamp=time.time()):    
    pass
```

Is it reasonable that any default expression that is a function call 
should be detected as a dangerous-default-value / dangerous-default-expression ?

P.S. pylint is great :-)    

### Desired solution

Any default expression that is a function call is detected as a dangerous-default-value / dangerous-default-expression

**Comment by Pierre-Sassoulas** [comment]
Duplicate of https://github.com/pylint-dev/pylint/issues/4659 which have an open PR that need take over (https://github.com/pylint-dev/pylint/pull/8553)
